falcon/media/json.py

```diff
@@ -12,19 +12,88 @@ class JSONHandler(BaseHandler):
 
     This handler uses Python's :py:mod:`json` by default, but will
     use :py:mod:`ujson` if available.
+
+    Keyword Arguments:
+        dumps (func): the function to use when serializing JSON responses
+
+        loads (func): the function to use when deserializing JSON requests
+
+        dumps_kwargs (dict): keyword arguments passed to ``dumps`` when
+            serializing to JSON.  ``ensure_ascii`` is set to ``false`` by
+            default.  To override this setting, pass ``{ensure_ascii=True}``
+            explicitly. You can completely disable extra parameters by passing
+            ``False``.
+
+        loads_kwargs (func): keyword arguments to pass to ``loads`` when
+            deserializing from JSON.  You can completely disable extra
+            parameters by passing ``False``.
+
+    You can override the JSON library used by changing the ``dumps`` and
+    ``loads`` functions::
+
+        import falcon
+        from falcon import media
+        import rapidjson
```

Comment:
IMHO we could consider testing with ``orson`` as well. One, its author @ijl came here to share insights on the PR, and two, ``orson`` seems to be one the fastest kids on block indeed. Not to mention some Falcon maintainers have closet sympathy for Rust :smiling_imp:

Reply:
👍

Reply:
For reference: https://github.com/ijl/orjson

Reply:
Actually, after thinking about this some more and looking at mujson's implementation more closely, I think we may want to just list it as something people might want to look at (and add it to our tests), and just stick with python-rapidjson in this example. mujson's approach may not be as straightforward and predictable as some people may like. orjson is super fast, but unfortunately not fully compliant as a drop-in replacement to stdlib json.

So, to summarize, we demonstrate python-rapidjson in this example, and then list these in the docs (as well as include them in a few smoke tests):

* orjson (fastest, but doesn't support some of the features provided by stdlib)
* python-rapidjson
* mujson (does not currently include orjson in the mix)

Plus we will add ujson to our smoke tests since we previously supported it, although it is not well-maintained these days:

* ujson

Reply:
Done! Thanks for the feedback